lib/disk/modules/AzureDiskCommon.rb

```diff
@@ -84,7 +85,16 @@ def blob_read(start_byte, length)
       options[:date] = @snapshot if @snapshot
       ret = @storage_acct.get_blob_raw(@container, @blob, key, options)
     else
-      ret = @storage_disk_svc.get_blob_raw(@disk_name, @resource_group, options)
+      retries = 0
+      begin
+        ret = managed_disk.read(options)
+      rescue Azure::Armrest::ForbiddenException => err
+        raise err if retries > 0
+        $log.warn("#{@class}: blob_read: #{err}: acquiring new SAS URL")
```

Comment:
Shouldn't this retry logic be in the `ManagedDisk` object itself? I thought that was one of the main reasons for implementing that class. The clients should be insulated from all of this.

Reply:
Let me see if I can do that given the restrictions I found myself working under in the disk_service / snapshot_service classes and the included modules.